### PR TITLE
[AI] Documentation Refactor

### LINKED ISSUE #941: [Skill Template][TypeScript] Generate Skill Template solution
**Is your feature request related to a problem? Please describe.**
The Skill Template solution is developed in C# but not in TypeScript

**Describe the solution you'd like**
Have the Skill Template solution developed in TypeScript

**Describe alternatives you've considered**
The following tasks should be solved:
- [X] Create project structure for TypeScript
- [X] Implement TSLint
- [X] Migration of services
     - [X] Migrate cognitiveModels
     - [X] Migrate deploymentScripts
     - [X] Generate scripts for deploy skills with PowerShell 
     - [X] Generate resources 
     - [X] Migrate serviceClients
- [X] Migration of logic
     - [X] Migrate SkillTemplate and State
     - [X] Migrate dialogs
- [x] Migration of tests
- [ ] Migration of related documentation

### LINKED ISSUE #1034: [Virtual-Assistant][TS] Bot file generated by "deploy_bot.ps1" fails to write the LUIS endpoint to file
## Project

**Name**: 
Virtual Assistant

**Language**: 
TypeScript

## Description

`deploy_bot.ps1` for the VA fails to write the luis endpoint details to the `.bot` file. 

## To Reproduce

1. Follow the deployment steps
1. At the intial deployment step run `pwsh.exe -ExecutionPolicy Bypass -File DeploymentScripts\deploy_bot.ps1`
1. Check the generated `.bot` file

Luis endpoint will be missing. 

```json
...
      {
            "type": "luis",
            "name": <APP_NAME>,
            "appId": <APP_ID>,
            "authoringKey": <HASHED_AUTHORING_KEY>,
            "subscriptionKey": <HASHED_SUBSCRIPTION_KEY>,
            "version": "0.1",
            "region": "westus",
            "id": "general"
        },
...
```

## Expected behavior

`.bot` should include the luis endpoint just like the enterprise template version.

### PR DESCRIPTION
**Pull request by Batta32**
## Description

- Separate some `docs` folders in `common`, `csharp`, `typescript`
- Add `create a new skill` document
- Update `Virtual Assistant` documentation
- Add a table with the work in progress for the documentation between versions
- Unify the docs betwen `csharp` and `typescript`
- Fix some link references
- Fix tables structure

Close #1034.
Close #941.

## Testing Steps
1. Go to `AI/docs/skills` and `AI/docs/virtual-assistant`.
2. Check the new structure of the documentation.
3. Go to `AI/docs/virtual-assistant/typescript/gettingstarted.md`.
4. Check that the documentation is more verbose now.

## Checklist
- [x] I have updated related documentation

### CODE REVIEW

**Comment by darrenj** [comment]
Looks great, @ryanlengel can you do a quick pass as you did the most recent refactor. This is to better separate csharp versus typescript and make things clearer for non c# devs

**Comment by ryanisgrig** [comment]
This is awesome - thanks @Batta32

**Review by darrenj — APPROVED** [review]

File: templates/Skill-Template/src/csharp/SkillTemplate/Skills/Skill Template/readme.md [hunk 1]
```diff
@@ -1 +1 @@
-﻿Please refer to [Create a New Skill](https://github.com/Microsoft/AI/blob/master/docs/skills/create.md) for instructions on creating your skill.
\ No newline at end of file
+﻿Please refer to [Create a New Skill](../../../../../../../docs/skills/README.md#skills) for instructions on creating your skill.
```
Comment (ryanisgrig) [inline templates/Skill-Template/src/csharp/SkillTemplate/Skills/Skill Template/readme.md]: This should point to https://github.com/Microsoft/AI/blob/master/docs/skills/README.md#skills, this README is used in the deployed skill template.

**Review by ryanisgrig — CHANGES_REQUESTED** [review]

File: templates/Skill-Template/src/csharp/SkillTemplate/Skills/Skill Template/readme.md (hunk 1 above)
Comment (Batta32) [reply templates/Skill-Template/src/csharp/SkillTemplate/Skills/Skill Template/readme.md]: Hi @ryanlengel, thanks for your feedback!
You are right with that, I've already fixed it and pointed to that link

**Review by ryanisgrig — APPROVED** [review]

**Pull request merged**